### ISSUE TITLE
[Feature request] Adding support for IMPACT-Z and GENESIS1.3 version 4

### ISSUE THREAD

**Issue by vgrattoni** [slaclab/lume#1]
Hi,
I saw that your project supports IMPACT-T and GENESIS1.3 version 2. Is it possible to add the support for IMPACT-Z and [GENESIS 1.3 version 4](https://github.com/svenreiche/Genesis-1.3-Version4)?
I think it would be very useful for studying EEHG seeded FEL (and other advanced FEL schemes) with a realistic number of particles as the new version of GENESIS is enabling it. 
Thank you very much efforts

**Comment by ChristopherMayes** [comment]
@vgrattoni Yes, we are working on adding IMPACT-Z to lume-impact, and there has been work started with Genesis 1.3 version 4: https://github.com/slaclab/lume-genesis/tree/master/genesis4

https://github.com/slaclab/lume-genesis/blob/master/examples/Genesis4.ipynb
